agent/engine/docker_container_engine.go

```diff
@@ -168,7 +168,27 @@ var scratchCreateLock sync.Mutex
 
 // NewDockerGoClient creates a new DockerGoClient
 func NewDockerGoClient(clientFactory dockerclient.Factory, cfg *config.Config) (DockerClient, error) {
-	client, err := clientFactory.GetDefaultClient()
+	// Use 1.21 client if container metadata is enabled; otherwise use default
```

Comment:
Do not change the global default.  If you need a specific client version in your code, retrieve that client version from the factory.  Changing the default here has the potential to break many things.